soccer/src/soccer/strategy/agent/position/marker.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <rclcpp/rclcpp.hpp>
+
+#include <rj_geometry/segment.hpp>
+#include <rj_msgs/msg/robot_state.hpp>
+#include <rj_msgs/msg/world_state.hpp>
+
+#include "position.hpp"
+#include "rj_geometry/point.hpp"
+#include "role_interface.hpp"
+
+namespace strategy {
+
+/**
+ * Represents a Marker, a defensive role that targets an enemy robot
+ * and follows it around the field while it is on our side, blocking passes.
+ */
+class Marker : public RoleInterface {
```

Comment:
note the GH action warning here. check the "rule of five" under https://en.cppreference.com/w/cpp/language/rule_of_three. you can explicitly set all of those functions to `=default;`